raven/src/main/java/org/opencadc/raven/RavenInitAction.java

```diff
@@ -235,4 +282,91 @@ static MultiValuedProperties getConfig() {
             throw new IllegalStateException("invalid config: failed to load SQLGenerator: " + cname);
         }
     }
+
+    private class AvailabilityCheck implements Runnable {
+
+        private Set<StorageSite> sites;
+        private Map<URI, SiteState> siteStaties;
+        private Map<URI, Availability> siteAvailabilities;
+        
+        public AvailabilityCheck(Set<StorageSite> sites) {
+            this.sites = sites;
+            this.siteStaties = new HashMap<>(sites.size());
+            this.siteAvailabilities = new HashMap<>(sites.size());
+            try {
+                Context initialContext = new InitialContext();
+                initialContext.bind(JNDI_AVAILABILITY_KEY, this.siteAvailabilities);
+            } catch (NamingException e) {
+                throw new IllegalStateException(String.format("unable to bind {} to initial context: %s", 
+                                                JNDI_AVAILABILITY_KEY, e.getMessage()), e);
+            }
+        }
+
+        @Override
+        public void run() { 
+            while (true) {
+                for (StorageSite site: this.sites) {
+                    try {
+                        URI resourceID = site.getResourceID();
+                        SiteState siteState = this.siteStaties.get(resourceID);
+                        if (siteState == null) {
+                            siteState = new SiteState(true, 0);
+                        }
+                        boolean minDetail = siteState.isMinDetail();
+                        Availability availability = getAvailability(resourceID, minDetail);
+                        siteState.available = availability.isAvailable();
+                        this.siteStaties.put(resourceID, siteState);
+                        this.siteAvailabilities.put(resourceID, availability);
+                        Context initContext = new InitialContext();
+                        initContext.rebind(JNDI_AVAILABILITY_KEY, this.siteAvailabilities);
+                        log.info(String.format("availability check [%s] for %s [%s]", 
```

Comment:
I would log successful checks at debug and failed checks at warn. It would be nice to log "return to success" at info but it's probably more pain than it's worth.

Reply:
Updated the logging to debug at success and warn for failures.